packages/application-config/schema.json

```diff
@@ -0,0 +1,133 @@
+      "description": "The URL of the Merchant Center API. We usually recommend to use the `cloudIdentifier` option to avoid possible typos.",
+      "type": "string"
+    },
+    "env": {
+      "description": "This is an object of keys that represent different environments (for example `production`). The environment used depends on the environment variable MC_APP_ENV. If MC_APP_ENV isn't set then NODE_ENV will be used, if it's not set then it defaults to \"development\"",
+      "type": "object",
+      "properties": {
+        "production": {
+          "description": "Configuration for production only",
+          "type": "object",
+          "properties": {
+            "url": {
+              "description": "The URL where the Custom Application is hosted",
+              "type": "string"
+            },
+            "cdnUrl": {
+              "description": "The URL where the Custom Application static assets are hosted, like an external CDN. If the static assets are hosted alongside the Custom Application, you can omit this option and the Custom Application URL will be used instead.",
+              "type": "string"
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "url"
+          ]
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "production"
+      ]
+    },
+    "additionalEnv": {
+      "description": "Additional environment values that are injected in the application context",
+      "type": "object"
+    },
+    "headers": {
+      "description": "Configuration for HTTP headers",
+      "type": "object",
+      "properties": {
+        "csp": {
+          "description": "Configuration for the HTTP Content-Security-Policy header (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy)",
+          "type": "object",
+          "properties": {
+            "connect-src": {
+              "$ref": "#/definitions/cspDirective"
+            },
+            "font-src": {
+              "$ref": "#/definitions/cspDirective"
+            },
+            "img-src": {
+              "$ref": "#/definitions/cspDirective"
+            },
+            "script-src": {
+              "$ref": "#/definitions/cspDirective"
+            },
+            "style-src": {
+              "$ref": "#/definitions/cspDirective"
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "connect-src"
+          ]
+        },
+        "featurePolicies": {
+          "description": "Configuration for the HTTP Feature-Policy header (https://developer.mozilla.org/en-US/docs/Web/HTTP/Feature_Policy/Using_Feature_Policy)",
+          "type": "object"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "csp"
+      ]
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "name",
+    "entryPointUriPath",
+    "cloudIdentifier",
+    "env"
+  ]
```

Comment:
I see that `id` is not required, is it on purpose. I think every custom app will have a `ID`

Reply:
Good observation. The ID is not required (yet), it will be in future releases and when we (soon) implement a new workflow to register custom apps.